src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -312,25 +314,36 @@ InvokeCallbackForDatePattern
 Gets the ICU date pattern for the specified locale and EStyle and invokes the
 callback with the result.
 */
-bool InvokeCallbackForDatePattern(Locale& locale,
-                                  DateFormat::EStyle style,
+bool InvokeCallbackForDatePattern(const char* locale,
+                                  UDateFormatStyle style,
                                   EnumCalendarInfoCallback callback,
                                   const void* context)
 {
-    LocalPointer<DateFormat> dateFormat(DateFormat::createDateInstance(style, locale));
-    if (dateFormat.isNull())
+    UErrorCode err = U_ZERO_ERROR;
+    UDateFormat* pFormat = udat_open(UDAT_NONE, style, locale, nullptr, 0, nullptr, 0, &err);
+    UDateFormatHolder formatHolder(pFormat, err);
+
+    if (U_FAILURE(err))
         return false;
 
-    // cast to SimpleDateFormat so we can call toPattern()
-    SimpleDateFormat* sdf = dynamic_cast<SimpleDateFormat*>(dateFormat.getAlias());
-    if (sdf == NULL)
+    UErrorCode ignore = U_ZERO_ERROR;
+    int32_t patternLen = udat_toPattern(pFormat, false, nullptr, 0, &ignore);
+
+    UChar* pattern = (UChar*)calloc(patternLen + 1, sizeof(UChar));
```

Comment:
I will change this to use std::vector like I do the other places (long term maybe we move off off std::vector and on to some custom thing, but might as well do the same things we do everywhere else).